src/core/ext/filters/load_reporting/load_reporting.c

```diff
@@ -42,9 +42,13 @@ static bool maybe_add_load_reporting_filter(grpc_exec_ctx *exec_ctx,
                                             void *arg) {
   const grpc_channel_args *args =
       grpc_channel_stack_builder_get_channel_arguments(builder);
-  if (is_load_reporting_enabled(args)) {
-    return grpc_channel_stack_builder_prepend_filter(
-        builder, (const grpc_channel_filter *)arg, NULL, NULL);
+  const grpc_channel_filter *filter = arg;
+  grpc_channel_stack_builder_iterator *it = grpc_channel_stack_builder_iterator_find(builder, filter->name);
+  const bool already_has_load_reporting_filter = !grpc_channel_stack_builder_iterator_is_end(it);
+  grpc_channel_stack_builder_iterator_destroy(it);
+  if (is_load_reporting_enabled(args) && !already_has_load_reporting_filter) {
```

Comment:
The test's initial call to `::testing::InitGoogleTest()` will trigger all (internal) module initializers. One of them is `InitGRPC()`, which will register the internal version of the load reporting plugin.

The OSS version, from `grpc_init()` is, in this case, initialized later as part of instantiating `ServerCompletionQueue` because [`CompletionQueue` inherits from `GrpcLibraryCodegen`](https://github.com/grpc/grpc/blob/f4af6eb2f44e7a9fa08e5e427a5e5f9d1ba2acd2/include/grpc%2B%2B/impl/codegen/completion_queue.h#L86), whose job is precisely to [call `grpc_init()`](https://github.com/grpc/grpc/blob/5253c8f9a899450397a5e46e4923d01ac9a66a27/include/grpc%2B%2B/impl/codegen/grpc_library.h#L44). `ServerCompletionQueue` is not instantiated until the test needs to create a server instance.

Basically, internal happens first because the (internal) testing framework initializes internal modules first.